### PR TITLE
misc node:zlib fixes [v2]

### PR DESCRIPTION
**Pull request by nektro**
rebase of https://github.com/oven-sh/bun/pull/13784
more to come but there's a few bug fixes in here i wanted to get up in the meantime

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit 6b2e2527a9e44ad3f05afb7fbb45247a3b03b896 is building: [`#3497`](https://buildkite.com/bun/bun/builds/3497)

File: src/js/node/zlib.ts [hunk 1]
```diff
@@ -45,34 +45,59 @@ function Base(method, options) {
...
+  get: function () {
+    return this[kHandle].finishFlush;
+  },
+});
+ObjectDefineProperty(ZlibBase.prototype, "_defaultFullFlushFlag", {
+  get: function () {
+    return this[kHandle].fullFlush;
+  },
+});
+ObjectDefineProperty(ZlibBase.prototype, "_maxOutputLength", {
```
Comment (Jarred-Sumner) [inline src/js/node/zlib.ts]: Does this match node? Are these fields read-only and non-enumerable and non-configurable in node?

**Comment by Jarred-Sumner** [comment]
Are there any new node tests that pass?

File: src/js/node/zlib.ts (hunk 1 above)
Comment (nektro) [reply src/js/node/zlib.ts]: not exactly but left a doc comment and a TODO to come back as well as made sure they were enumerable

**Comment by nektro** [comment]
didnt check new tests but this does fix a number of bugs already:

- fixes a memory leak in crc32
- fixes a crash when brotli params are undefined
- fixes an error when zlib dictionary is undefined
- adds observable values for internal fields packages may be checking
- fixes internal class names packages may be checking

**Pull request merged**